scripts/test/Muon/elemental_analysis/elemental_analysis_test.py

```diff
@@ -0,0 +1,502 @@
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.elemental_analysis.ElementalAnalysisGui.get_color')
+    def test_that_add_element_lines_will_call_get_color(self, mock_get_color):
+        self.gui._add_element_lines('Cu')
+        self.assertEqual(mock_get_color.call_count, 1)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.elemental_analysis.ElementalAnalysisGui._plot_line')
+    def test_that_add_element_lines_will_call_plot_line(self, mock_plot_line):
+        data = {'line1': 10.0, 'line2': 20.0, 'line3': 30.0}
+        self.gui._add_element_lines('Cu', data)
+        self.assertEqual(mock_plot_line.call_count, 3)
+        call_list = [mock.call(gen_name('Cu', 'line3'), 30.0, 'C0', 'Cu'),
+                     mock.call(gen_name('Cu', 'line2'), 20.0, 'C0', 'Cu'),
+                     mock.call(gen_name('Cu', 'line1'), 10.0, 'C0', 'Cu')]
+        mock_plot_line.assert_has_calls(call_list, any_order=True)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.elemental_analysis.ElementalAnalysisGui._rm_line')
+    def test_remove_element_lines_does_nothing_if_element_not_in_element_lines(self, mock_rm_line):
+        self.gui.element_lines['H'] = ['alpha', 'beta']
+        self.gui._remove_element_lines('He')
+        self.assertEqual(mock_rm_line.call_count, 0)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.elemental_analysis.ElementalAnalysisGui._rm_line')
+    def test_remove_element_lines_removes_all_values_for_a_given_element(self, mock_rm_line):
+        self.gui.element_lines['H'] = ['alpha', 'beta']
+        self.gui._remove_element_lines('H')
+        self.assertEqual(mock_rm_line.call_count, 2)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.elemental_analysis.MultiPlotWindow')
+    def test_load_run_opens_new_plot_window_if_none_open(self, mock_MultiPlotWindow):
+        self.gui.add_detector_to_plot = mock.Mock()
+        self.gui.load_run('GE1', '2695')
+        self.assertEqual(mock_MultiPlotWindow.call_count, 1)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.elemental_analysis.MultiPlotWindow')
+    def test_load_run_does_not_open_new_plot_window_if_one_is_open(self, mock_MultiPlotWindow):
+        self.gui.add_detector_to_plot = mock.Mock()
+        self.gui.plot_window = MultiPlotWindow(str('2695'))
+        self.gui.load_run('GE1', 2695)
+        self.assertEqual(mock_MultiPlotWindow.call_count, 0)
+
+    def test_loading_finished_returns_nothing_if_no_run_loaded(self):
+        self.gui.load_widget.last_loaded_run = mock.Mock()
+        self.gui.load_widget.last_loaded_run.return_value = None
+        self.gui.detectors = mock.Mock()
+        self.gui.detectors.detectors = [mock.Mock(), mock.Mock(), mock.Mock(), mock.Mock()]
+        for i in self.gui.detectors.detectors:
+            i.isChecked.return_value = True
+        self.gui.plot_window = mock.Mock()
+        self.gui.plotting = mock.Mock()
+
+        self.gui.loading_finished()
+        self.assertEqual(self.gui.plotting.remove_subplot.call_count, 0)
```

Comment:
should we not be checking add_subplot?

Reply:
The function `loading_finished` calls `remove_subplot`:
https://github.com/mantidproject/mantid/blob/e383f53ca60873572815ef90ea75d596fc71b153/scripts/Muon/GUI/ElementalAnalysis/elemental_analysis.py#L239